value/value.go

```diff
@@ -268,28 +265,17 @@ func (m *Map) Less(rhs *Map) bool {
 
 // Get returns the (Field, true) or (nil, false) if it is not present
 func (m *Map) Get(key string) (*Field, bool) {
-	if m.index == nil {
-		m.index = map[string]int{}
-		for i := range m.Items {
-			m.index[m.Items[i].Name] = i
+	for i := len(m.Items) - 1; i >= 0; i-- {
```

Comment:
You may be optimizing for the one-shot case at the expense of the repeated case.